sdk/python/feast/feature_set.py

```diff
@@ -361,8 +388,9 @@ def infer_fields_from_df(
 
         # Create dictionary of fields that will not be inferred (manually set)
         provided_fields = OrderedDict()
+        fields = self._entities_and_features_to_fields(entities, features)
 
-        for field in entities + features:
+        for field in fields:
```

Comment:
You are testing all of the fields twice...

Reply:
Ah by the way, can you elaborate more on this? what do you mean?

Is it the `is_instance` check?

Reply:
Fields contains all the features and entities. You are looping over it twice. I dont think you need `entities_and_features_to_fields`. You can just loop over entities and features.

Reply:
The mypy convention was because it cannot handle the `entities + features` syntax where we process both of them at the same time, separating into two iteration also require both the feature and entity to be aggregated, which would create the same number of iteration.
So the field converter was created. 

I don't think it would affect the performance, as field and entity is limited, the maximum of N in the O(N) is very less (less than 1000 feature + entity) also 2 converter iteration is very fast process

Reply:
Over time we will move away from Fields because we want to treat entities and features separately. Which is why I would prefer to do validation and building of this `dict` from within the method itself, and not to create a new method. For the time being this proposal is ok

Can we at least rename this `entities_and_features_to_fields` so that it is more descriptive?
`_create_field_list` or something?